Ira/my-app-video/src/components/filmPage/FilmPage.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import {RouteComponentProps} from "react-router-dom";
+import {Header} from "../layout/header";
+import {FullInfoFilmCard} from "../fullInfoFilmCard";
+import {Footer} from "../layout/footer";
+import {FilmList} from "../filmList";
+import {Button} from "../shared/button/Button";
+import Pagination from "../pagination";
+import {filmPageConnectProps} from "./index";
+import {queryString} from "../../constants/queryString";
+import {SortBy} from "../sortingSection/SortingSection";
+import {SearchBy} from "../shared/searchFilmForm/SearchFilmForm";
+import './styles.scss';
+
+type FilmPageProps = RouteComponentProps<{ id: string }> & filmPageConnectProps;
+
+export class FilmPage extends React.Component<FilmPageProps> {
+  componentDidMount() {
+    this.initPage();
+  }
+
+  componentDidUpdate(prevProps: Readonly<FilmPageProps>) {
+    // eslint-disable-next-line no-mixed-operators
+    if (this.props.history.action !== 'PUSH' && this.props.location !== prevProps.location || this.props.location.pathname !== prevProps.location.pathname) {
```

Comment:
hard to read. may be potential issues. Please wrap conditions into a brackets. now it will work like this:  `(this.props.history.action !== 'PUSH' && this.props.location !== prevProps.location) || this.props.location.pathname !== prevProps.location.pathname`. I am not sure this is what your really need.